### PR TITLE
cli: bump sucrase and disable ES transforms

### PR DESCRIPTION
**Pull request by Rugvip**
Sucrase has just added an option to disable ECMAScript transforms, which is useful for us because we only target environment that already has support for the syntax that was being transformed. This does depend on #6889 though as Node.js v12 doesn't have full support.

The main chunk shrunk about 3.3% while some of the smaller chunks lost upwards of 20%. Skipping the transforms should also make runtime execution faster.

The reason for enabling `useDefineForClassFields` is described in https://github.com/alangpierce/sucrase#transforms, and the overall purpose of that flag is described in https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#the-usedefineforclassfields-flag-and-the-declare-property-modifier. TL;DR is that TypeScript is not compatible with ECMAScript in its handling of class fields and is slowly migrating over, and sucrase used to transform them to be compatible with TypeScript.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -25360,7 +25360,7 @@ subscriptions-transport-ws@^0.9.18, subscriptions-transport-ws@^0.9.19:
     symbol-observable "^1.0.4"
     ws "^5.2.0 || ^6.0.0 || ^7.0.0"
 
-sucrase@^3.18.0, sucrase@^3.18.2:
+sucrase@^3.18.0:
```
Comment (freben) [inline yarn.lock:25363]: Maybe merge this block with the one below

File: packages/cli/src/lib/bundler/transforms.ts [hunk 2]
```diff
@@ -39,6 +39,7 @@ export const transforms = (options: TransformOptions): Transforms => {
       loader: require.resolve('@sucrase/webpack-loader'),
       options: {
         transforms: ['typescript', 'jsx', ...extraTransforms],
+        disableESTransforms: true,
```
Comment (freben) [inline packages/cli/src/lib/bundler/transforms.ts:42]: This flag sounds so coarse. ALL of them! :$

**Review by freben — APPROVED** [review]

File: yarn.lock (hunk 1 above)
Comment (Rugvip) [reply yarn.lock:25363]: Can't, the jest plugin still pulls in the old version :<

File: packages/cli/src/lib/bundler/transforms.ts (hunk 2 above)
Comment (Rugvip) [reply packages/cli/src/lib/bundler/transforms.ts:42]: Sucrase is very opinionated, so this fits with that :grin:. It keeps the implementation and maintenance simpler.

File: yarn.lock (hunk 1 above)
Comment (freben) [reply yarn.lock:25363]: Hm maybe I'm blind, they looked semver-compatible

**Pull request merged**